src/components/Notification.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { SyntheticEvent } from 'react';
+import { EnrichedActivity, EnrichedUser, NotificationActivityEnriched, UR } from 'getstream';
+import { TFunction } from 'i18next';
+
+import { Avatar } from './Avatar';
+import { AvatarGroup } from './AvatarGroup';
+import { AttachedActivity } from './AttachedActivity';
+import { Dropdown } from './Dropdown';
+import { Link } from './Link';
+import { humanizeTimestamp, useOnClickUser, userOrDefault, OnClickUserHandler } from '../utils';
+import { DefaultUT, DefaultAT, useTranslationContext, FeedManager } from '../Context';
+
+export type NotificationProps<
+  UT extends DefaultUT = DefaultUT,
+  AT extends DefaultAT = DefaultAT,
+  CT extends UR = UR,
+  RT extends UR = UR,
+  CRT extends UR = UR
+> = {
+  /** The activity group to display in this notification */
+  activityGroup: NotificationActivityEnriched<UT, AT, CT, RT, CRT>;
+  /** Callback to call when clicking on a notification */
+  onClickNotification?: (activityGroup: NotificationActivityEnriched<UT, AT, CT, RT, CRT>) => void;
+  /** Callback to call when clicking on a user in the notification */
+  onClickUser?: OnClickUserHandler<UT>;
+  /** Callback to mark a notification as read, if not supplied the dropdown used to mark as read will not be shown */
+  onMarkAsRead?: FeedManager<UT, AT, CT, RT, CRT>['onMarkAsRead'];
+};
+
+const getUsers = <
+  UT extends DefaultUT = DefaultUT,
+  AT extends DefaultAT = DefaultAT,
+  CT extends UR = UR,
+  RT extends UR = UR,
+  CRT extends UR = UR
+>(
+  activities: Array<EnrichedActivity<UT, AT, CT, RT, CRT>>,
+) => activities.map((item) => userOrDefault<UT>(item.actor));
+
+const getHeaderText = (t: TFunction, activitiesLen: number, verb: string, actorName: string, activityVerb: string) => {
+  if (activitiesLen === 1) {
+    if (verb === 'like') {
+      return t('{{ actorName }} liked your {{ activityVerb }}', { actorName, activityVerb });
+    }
+    if (verb === 'repost') {
+      return t('{{ actorName }} reposted your {{ activityVerb }}', { actorName, activityVerb });
+    }
+    if (verb === 'follow') {
+      return t('{{ actorName }} followed you', { actorName });
+    }
+    if (verb === 'comment') {
+      return t('{{ actorName }} commented on your {{ activityVerb }}', { actorName, activityVerb });
+    }
+    console.warn('No notification styling found for your verb, please create your own custom Notification group.');
+    return '';
```

Comment:
Could probably use `switch` here (and for the rest) to make it look a bit cleaner.